implementations/rust/ockam/ockam_entity/src/channel/secure_channel_worker.rs

```diff
@@ -507,6 +522,38 @@ impl<I: Identity, T: TrustPolicy> SecureChannelWorker<I, T> {
 
         Ok(())
     }
+
+    async fn handle_api(
+        &mut self,
+        ctx: &mut Context,
+        msg: Routed<<Self as Worker>::Message>,
+        s: State<I, T>,
+    ) -> Result<()> {
+        let return_route = msg.return_route();
+
+        let msg = SecureChannelApiRequest::decode(msg.payload())?;
```

Comment:
This is something that I do a lot in the `ockam` crate as well. I'm wondering if maybe we could just add a function to `TransportMessage` that will decode the payload and return `T`

Reply:
@spacekookie nice idea, added as issue #2298